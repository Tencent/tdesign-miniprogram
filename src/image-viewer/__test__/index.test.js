import path from 'path';
import simulate from 'miniprogram-simulate';

describe('image-viewer', () => {
  const imageViewer = load(path.resolve(__dirname, `../image-viewer`), 't-image-viewer');

  it(`: style && customStyle`, async () => {
    const id = simulate.load({
      template: `<t-image-viewer class="imageViewer" visible style="{{style}}" customStyle="{{customStyle}}"></t-image-viewer>`,
      usingComponents: {
        't-image-viewer': imageViewer,
      },
      data: {
        style: 'color: red',
        customStyle: 'font-size: 9px',
      },
    });
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));
    const $imageViewer = comp.querySelector('.imageViewer >>> .t-image-viewer');
    // expect(comp.toJSON()).toMatchSnapshot();
    if (VIRTUAL_HOST) {
      expect(
        $imageViewer.dom.getAttribute('style').includes(`${comp.data.style}; ${comp.data.customStyle}`),
      ).toBeTruthy();
    } else {
      expect($imageViewer.dom.getAttribute('style').includes(`${comp.data.customStyle}`)).toBeTruthy();
    }
  });

  it(':base', async () => {
    const id = simulate.load({
      template: `<t-image-viewer visible id="base" images="{{images}}" />`,
      data: {
        images: ['a.png', 'b.png'],
      },
      usingComponents: {
        't-image-viewer': imageViewer,
      },
    });
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    const $image = comp.querySelector('#base >>> .t-image-viewer__image');

    $image.dispatchEvent('load', { detail: { width: 100, height: 100 } });
    await simulate.sleep();

    expect($image.toJSON()).toMatchSnapshot();

    $image.dispatchEvent('load', { detail: { width: 1000, height: 1000 } });
    await simulate.sleep();

    expect($image.toJSON()).toMatchSnapshot();

    $image.dispatchEvent('load', { detail: { width: 800, height: 1000 } });
    await simulate.sleep();

    expect($image.toJSON()).toMatchSnapshot();
  });

  it(':event', async () => {
    const handleClose = jest.fn();
    const handleDelete = jest.fn();
    const id = simulate.load({
      template: `<t-image-viewer
        id="base"
        visible
        closeBtn
        deleteBtn
        bind:delete="handleDelete"
        bind:close="handleClose"
        images="{{images}}" />`,
      data: {
        images: ['a.png', 'b.png'],
      },
      methods: {
        handleClose,
        handleDelete,
      },
      usingComponents: {
        't-image-viewer': imageViewer,
      },
    });
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    // close
    const $close = comp.querySelector('#base >>> .t-image-viewer__nav-close');

    $close.dispatchEvent('tap');
    await simulate.sleep();

    expect(handleClose).toHaveBeenCalled();

    // delete
    const $delete = comp.querySelector('#base >>> .t-image-viewer__nav-delete');

    $delete.dispatchEvent('tap');
    await simulate.sleep();

    expect(handleDelete).toHaveBeenCalled();
  });

  it(':width', () => {});
});
