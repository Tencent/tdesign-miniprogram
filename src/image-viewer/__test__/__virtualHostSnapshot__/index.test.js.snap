// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image-viewer :base 1`] = `
<t-image
  class="t-image-viewer__image"
  data-index="{{0}}"
  tClass="t-image--external"
  bind:load="onImageLoadSuccess"
>
  <wx-view
    class="class t-class t-loading t-loading--horizontal"
    style="width: 44rpx; height: 44rpx; color: inherit;"
  >
    <wx-view
      ariaLabel="加载中"
      ariaRole="img"
      class="t-class-indicator t-loading__spinner t-loading__spinner--dots "
      style="width: 44rpx; height: 44rpx; color: inherit;  animation-duration: 0.8s; animation-play-state: running;"
    >
      <wx-view
        class="class t-class t-loading t-loading--horizontal"
        style="width: 44rpx; height: 44rpx;  color: inherit"
      >
        <wx-view
          ariaLabel="加载中"
          ariaRole="img"
          class="t-class-indicator t-loading__spinner t-loading__spinner--dots "
          style="width: 44rpx; height: 44rpx; color: inherit ; animation-duration: 0.8s; animation-play-state: running;"
        >
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0s; animation-play-state: running;"
          />
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0.26666666666666666s; animation-play-state: running;"
          />
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0.5333333333333333s; animation-play-state: running;"
          />
        </wx-view>
        <wx-view
          ariaHidden="{{true}}"
          ariaLabel=""
          class="t-loading__text t-loading__text--horizontal t-class-text"
        />
      </wx-view>
    </t-loading>
  </wx-view>
  <wx-image
    ariaHidden="{{true}}"
    ariaLabel=""
    class="class t-class t-image t-image--shape-square"
    hidden="{{true}}"
    id="image"
    lazyLoad="{{true}}"
    mode="aspectFit"
    showMenuByLongpress="{{false}}"
    src="a.png"
    style=""
    webp="{{false}}"
    bind:error="onLoadError"
    bind:load="onLoaded"
  />
</t-image>
`;

exports[`image-viewer :base 2`] = `
<t-image
  class="t-image-viewer__image"
  data-index="{{0}}"
  tClass="t-image--external"
  bind:load="onImageLoadSuccess"
>
  <wx-view
    class="class t-class t-loading t-loading--horizontal"
    style="width: 44rpx; height: 44rpx; color: inherit;"
  >
    <wx-view
      ariaLabel="加载中"
      ariaRole="img"
      class="t-class-indicator t-loading__spinner t-loading__spinner--dots "
      style="width: 44rpx; height: 44rpx; color: inherit;  animation-duration: 0.8s; animation-play-state: running;"
    >
      <wx-view
        class="class t-class t-loading t-loading--horizontal"
        style="width: 44rpx; height: 44rpx;  color: inherit"
      >
        <wx-view
          ariaLabel="加载中"
          ariaRole="img"
          class="t-class-indicator t-loading__spinner t-loading__spinner--dots "
          style="width: 44rpx; height: 44rpx; color: inherit ; animation-duration: 0.8s; animation-play-state: running;"
        >
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0s; animation-play-state: running;"
          />
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0.26666666666666666s; animation-play-state: running;"
          />
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0.5333333333333333s; animation-play-state: running;"
          />
        </wx-view>
        <wx-view
          ariaHidden="{{true}}"
          ariaLabel=""
          class="t-loading__text t-loading__text--horizontal t-class-text"
        />
      </wx-view>
    </t-loading>
  </wx-view>
  <wx-image
    ariaHidden="{{true}}"
    ariaLabel=""
    class="class t-class t-image t-image--shape-square"
    hidden="{{true}}"
    id="image"
    lazyLoad="{{true}}"
    mode="aspectFit"
    showMenuByLongpress="{{false}}"
    src="a.png"
    style=""
    webp="{{false}}"
    bind:error="onLoadError"
    bind:load="onLoaded"
  />
</t-image>
`;

exports[`image-viewer :base 3`] = `
<t-image
  class="t-image-viewer__image"
  data-index="{{0}}"
  tClass="t-image--external"
  bind:load="onImageLoadSuccess"
>
  <wx-view
    class="class t-class t-loading t-loading--horizontal"
    style="width: 44rpx; height: 44rpx; color: inherit;"
  >
    <wx-view
      ariaLabel="加载中"
      ariaRole="img"
      class="t-class-indicator t-loading__spinner t-loading__spinner--dots "
      style="width: 44rpx; height: 44rpx; color: inherit;  animation-duration: 0.8s; animation-play-state: running;"
    >
      <wx-view
        class="class t-class t-loading t-loading--horizontal"
        style="width: 44rpx; height: 44rpx;  color: inherit"
      >
        <wx-view
          ariaLabel="加载中"
          ariaRole="img"
          class="t-class-indicator t-loading__spinner t-loading__spinner--dots "
          style="width: 44rpx; height: 44rpx; color: inherit ; animation-duration: 0.8s; animation-play-state: running;"
        >
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0s; animation-play-state: running;"
          />
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0.26666666666666666s; animation-play-state: running;"
          />
          <wx-view
            class="t-loading__dot"
            style="animation-duration: 0.8s; animation-delay:0.5333333333333333s; animation-play-state: running;"
          />
        </wx-view>
        <wx-view
          ariaHidden="{{true}}"
          ariaLabel=""
          class="t-loading__text t-loading__text--horizontal t-class-text"
        />
      </wx-view>
    </t-loading>
  </wx-view>
  <wx-image
    ariaHidden="{{true}}"
    ariaLabel=""
    class="class t-class t-image t-image--shape-square"
    hidden="{{true}}"
    id="image"
    lazyLoad="{{true}}"
    mode="aspectFit"
    showMenuByLongpress="{{false}}"
    src="a.png"
    style=""
    webp="{{false}}"
    bind:error="onLoadError"
    bind:load="onLoaded"
  />
</t-image>
`;
